File: etamen-backend/app/Modules/CarePlans/Infrastructure/Models/CarePlanMeal.php
<?php

namespace App\Modules\CarePlans\Infrastructure\Models;

use App\Modules\CarePlans\Domain\Enums\CarePlanMealType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class CarePlanMeal extends Model
{
    protected $fillable = [
        'care_plan_day_id',
        'meal_type',
        'title',
        'description',
        'calories',
        'protein_g',
        'carbs_g',
        'fat_g',
        'instructions',
        'sort_order',
        'is_required',
    ];

    protected function casts(): array
    {
        return [
            'meal_type' => CarePlanMealType::class,
            'calories' => 'integer',
            'protein_g' => 'decimal:2',
            'carbs_g' => 'decimal:2',
            'fat_g' => 'decimal:2',
            'sort_order' => 'integer',
            'is_required' => 'boolean',
        ];
    }

    public function day(): BelongsTo
    {
        return $this->belongsTo(CarePlanDay::class, 'care_plan_day_id');
    }

    public function mealLogs(): HasMany
    {
        return $this->hasMany(MealLog::class);
    }
}

File: etamen-flutter/lib/features/care_plans/domain/entities/care_plan_meal.dart
class CarePlanMeal {
  const CarePlanMeal({
    required this.id,
    required this.mealType,
    this.carePlanDayId,
    this.title,
    this.description,
    this.calories,
    this.proteinG,
    this.carbsG,
    this.fatG,
    this.instructions,
    this.sortOrder,
    this.isRequired = false,
  });

  final int id;
  final int? carePlanDayId;
  final MealType mealType;
  final String? title;
  final String? description;
  final int? calories;
  final double? proteinG;
  final double? carbsG;
  final double? fatG;
  final String? instructions;
  final int? sortOrder;
  final bool isRequired;
}

enum MealType {
  breakfast('breakfast'),
  snack1('snack_1'),
  lunch('lunch'),
  snack2('snack_2'),
  dinner('dinner'),
  snack3('snack_3'),
  other('other'),
  unknown('unknown');

  const MealType(this.wireValue);

  final String wireValue;

  static MealType fromWire(Object? value) {
    final normalized = value?.toString();
    return MealType.values.firstWhere(
      (item) => item.wireValue == normalized,
      orElse: () => MealType.unknown,
    );
  }
}
